ucp/utils.py

```diff
@@ -80,6 +81,50 @@ def get_closest_net_devices(gpu_dev):
     return net_dev
 
 
+def get_ucxpy_logger():
+    """
+    Get UCX-Py logger with custom formatting
+
+    Returns
+    -------
+    logger : logging.Logger
+        Logger object
+
+    Examples
+    --------
+    >>> logger = get_ucxpy_logger()
+    >>> logger.warning("Test")
+    [1585175070.2911468] [dgx12:1054] UCXPY  WARNING Test
+    """
+
+    _level_enum = logging.getLevelName(os.getenv("UCXPY_LOG_LEVEL", "WARNING"))
+    logger = logging.getLogger("ucx")
+
+    # Avoid duplicate logging
+    logger.propagate = False
+
+    class LoggingFilter(logging.Filter):
+
+        def filter(self, record):
```

Comment:
Can you elaborate on why we need this filter ?  It's always returning `True` so is it filtering anything ?

Reply:
it returns `True` because we want to apply it for all logging messages. It could return `False` depending on some condition to avoid applying that to certain messages. For reference: https://docs.python.org/3/library/logging.html#logging.Logger.filter .